subsys/logging/log_backend_rtt.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2018 Workaround GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <logging/log_backend_rtt.h>
+#include <logging/log_core.h>
+#include <logging/log_msg.h>
+#include <logging/log_output.h>
+#include <device.h>
+#include <rtt/SEGGER_RTT.h>
+
+static int buffer_out(u8_t *data, size_t length, void *ctx)
+{
+	unsigned int key;
+
+	key = irq_lock();
```

Comment:
it is risky, CONFIG_LOG_BACKEND_RTT_BUFFER_SIZE is configurable so there is no control on critical section length. Checkout SEGGER_RTT_Conf.h file. There we can provide lock/unlock macros. If mutex is added there then here and in rtt_console.c SEGGER_RTT_Write relying on internal lock.